tests/test_riskpool_lifecycle.py

```diff
@@ -293,6 +293,295 @@ def test_suspend_resume(
 
     assert bundle2Id == bundleId + 1
 
+def test_suspend_archive(
+    instance: GifInstance, 
+    testCoin,
+    gifTestProduct: GifTestProduct, 
+    riskpoolKeeper: Account,
+    owner: Account,
+    capitalOwner: Account
+):
+    riskpool = gifTestProduct.getRiskpool().getContract()
+    riskpoolId = riskpool.getId()
+
+    instanceService = instance.getInstanceService()
+    instanceOperatorService = instance.getInstanceOperatorService()
+    assert instanceService.getComponentState(riskpoolId) == 3
+
+    initialFunding = 10000
+    bundleOwner = riskpoolKeeper
+    fund_riskpool(instance, owner, capitalOwner, riskpool, bundleOwner, testCoin, initialFunding)
+
+    riskpool.bundles() == 1
+    bundle = riskpool.getBundle(0)
+    print(bundle)
+
+    (
+        bundleId,
+        riskpoolId,
+        tokenId,
+        state,
+        filter,
+        capital,
+        lockedCapital,
+        balance,
+        createdAt,
+        updatedAt
+    ) = bundle
+
+    # check bundle values with expectation
+    assert bundleId == 1
+    assert riskpoolId == riskpool.getId()
+    assert owner != riskpool.owner()
+    print(riskpool.owner())
+
+    # verify that attributes directly from riskpool match with 
+    # bundle attributes from instance service
+    bundleFromCore = instanceService.getBundle(bundleId).dict()
+    print(bundleFromCore)
+
+    assert bundleFromCore["id"] == bundleId
+    assert bundleFromCore["riskpoolId"] == riskpoolId
+    assert bundleFromCore["tokenId"] == tokenId
+    assert bundleFromCore["state"] == state
+    assert bundleFromCore["filter"] == filter
+    assert bundleFromCore["capital"] == capital
+    assert bundleFromCore["lockedCapital"] == lockedCapital
+    assert bundleFromCore["balance"] == balance
+    assert bundleFromCore["createdAt"] == createdAt
+    assert bundleFromCore["updatedAt"] == updatedAt
+
+    componentOwnerService = instance.getComponentOwnerService()
+
+    # ensure that component owner may not archive riskpool

```

Comment:
comment mentions only component owner, also include instance operator in comment so it covers both brownie.reverts below